### ISSUE TITLE
Support HTTP Basic Authentication in project creation by URL download

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#217]
_Original author: mcnamara.tim@gmail.com (November 15, 2010 22:09:42)_

<b>What steps will reproduce the problem?</b>
1. Start project with https://demogy.service-now.com/incident.do?CSV
2. Receive 401 warning

<b>What is the expected output? What do you see instead?</b>

There are a few possible fixes. First, we could provide an error message with more detail for non-technical users. E.g. on a 401 error say that credentials are needed, rather than simply failing. Secondly, the product could provide somewhere to add in credentials.

_Original issue: http://code.google.com/p/google-refine/issues/detail?id=217_

**Comment by magdmartin** [comment]
works in 2.6-beta however it generate the following error if the password contains specific charactere like % with the index of the charactere.
````
Error uploading data
Malformed escape pair at index 
````

I tried escaping with a \ and made things worse
````
Error uploading data
Illegal character in authority at index 8: https://....
````

**Comment by tfmorris** [comment]
I'd have to do more research into how things are meant to be encoded, but a backslash (\) definitely won't help since it's not used in any of the URL encoding.

You could try replacing % with it's percent encoded equivalent (%25), but that's just a shot in the dark.

**Comment by antoine2711** [comment]
@tfmorris: so, if I understand, here, we have a request for a « Basic Authentication over HTTPS » on IMPORTATION to create a project?

Do this needs to be tested to see if it's still a problem? Do we keep the scope to CSV import or do we enlarge it to XML, and JSON (and maybe others…)?

Regards, Antoine

**Comment by thadguidry** [comment]
@tfmorris Yes, we talked a bit about this in various `fetch urls` issues and recently in #2379  (sigh...we have too many scattered issues regarding HTTP/S support)  The [`fetch urls` label](https://github.com/OpenRefine/OpenRefine/labels/fetch%20urls) is probably a good summary of the various HTTP/S issues either with Fetch operation, Importers, or Exporters.  In essence, to me as I mentioned [in my comment](https://github.com/OpenRefine/OpenRefine/issues/2379#issuecomment-603935878) on #2379 I think we just need to move to a higher level HTTP library and then provide better dialogs to assist the user in filling out the various parameters available.

**Comment by tfmorris** [comment]
I should have read the thread more carefully before replying. The last few comments don't have anything to do with @magdmartin's issue. 

The password in the header needs to be base64 encoded and it isn't currently. The feature also needs documentation and tests.

**Comment by wetneb** [comment]
Since we do support HTTPS already, this is really only about support for HTTP Basic Auth, so I have renamed the issue.
